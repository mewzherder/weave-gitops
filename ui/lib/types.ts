export enum PageRoute {
  Applications = "/applications",
  ApplicationDetail = "/application_detail",
}
