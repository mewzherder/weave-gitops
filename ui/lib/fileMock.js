// stub for ensuring tests don't try to load things like images
/* eslint-disable */
module.exports = "";
