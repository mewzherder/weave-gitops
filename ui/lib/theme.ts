// Typescript will handle type-checking/linting for this file
/* eslint-disable */
import { createMuiTheme } from "@material-ui/core";
import { createGlobalStyle, DefaultTheme } from "styled-components";
// @ts-ignore
import ProximaNovaBold from "url:../fonts/ProximaNovaBold.otf";
// @ts-ignore
import ProximaNovaRegular from "url:../fonts/ProximaNovaRegular.otf";
// @ts-ignore
import robotomono from "url:../fonts/roboto-mono-regular.woff";

const baseSpacingNumber = 16;

export const theme: DefaultTheme = {
  fontFamilies: {
    monospace: "'Roboto Mono', monospace",
    regular: "'proxima-nova', Helvetica, Arial, sans-serif",
  },
  fontSizes: {
    huge: "48px",
    extraLarge: "32px",
    large: "22px",
    normal: "16px",
    small: "14px",
    tiny: "12px",
  },
  colors: {
    black: "#1a1a1a",
    white: "#fff",
    primary: "#00b3ec",
    success: "#27AE60",
  },
  spacing: {
    // 16px
    base: `${baseSpacingNumber}px`,
    // 32px
    large: `${baseSpacingNumber * 2}px`,
    // 24px
    medium: `${baseSpacingNumber * 1.5}px`,
    none: "0",
    // 12px
    small: `${baseSpacingNumber * 0.75}px`,
    // 48px
    xl: `${baseSpacingNumber * 3}px`,
    // 8px
    xs: `${baseSpacingNumber * 0.5}px`,
    // 64px
    xxl: `${baseSpacingNumber * 4}px`,
    // 4px
    xxs: `${baseSpacingNumber * 0.25}px`,
  },
};

export default theme;

export const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: 'proxima-nova';
    src: url(${ProximaNovaRegular})
  }
  @font-face {
    font-family: 'proxima-nova';
    src: url(${ProximaNovaBold});
    font-weight: bold;
  }
  @font-face {
    font-family: 'Roboto Mono';
    src: url(${robotomono})
  }

  html,
  body {
    height: 100%;
    margin: 0;
  }

  #app {
    display: flex;
    flex-flow: column;
    height: 100%;
    margin: 0;
  }
  
  body {
    font-family: ${(props) => props.theme.fontFamilies.regular};
    font-size: ${theme.fontSizes.normal};
    padding: 0;
    margin: 0;
    min-width: fit-content;
  }
`;

export const muiTheme = createMuiTheme({
  typography: { fontFamily: "proxima-nova" },
});
