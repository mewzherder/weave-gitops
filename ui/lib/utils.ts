import qs from "query-string";
import { PageRoute } from "./types";

export const formatURL = (page: string, query: any = {}) => {
  return `${page}?${qs.stringify(query)}`;
};

export const getNavValue = (currentPage: any): PageRoute | boolean => {
  switch (currentPage) {
    case "applications":
    case "application_detail":
      return PageRoute.Applications;

    default:
      // The "Tabs" component of material-ui wants a bool
      return false;
  }
};
