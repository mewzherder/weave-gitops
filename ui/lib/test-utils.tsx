import { MuiThemeProvider } from "@material-ui/core";
import { createMemoryHistory } from "history";
import * as React from "react";
import { Router } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import theme, { muiTheme } from "./theme";

export function withTheme(element) {
  return (
    <MuiThemeProvider theme={muiTheme}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MuiThemeProvider>
  );
}

export function withContext(TestComponent, url: string) {
  const history = createMemoryHistory();
  history.push(url);
  return (
    <Router history={history}>
      <TestComponent />
    </Router>
  );
}
