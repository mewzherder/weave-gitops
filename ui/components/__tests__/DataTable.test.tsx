import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";
import { withTheme } from "../../lib/test-utils";
import DataTable from "../DataTable";

describe("DataTable", () => {
  describe("snapshots", () => {
    it("renders", () => {
      const rows = [
        {
          name: "my-cool-app",
          status: "Ready",
          lastUpdate: "2006-01-02T15:04:05-0700",
        },
        {
          name: "podinfo",
          status: "Ready",
          lastUpdate: "2006-01-02T15:04:05-0700",
        },
        {
          name: "nginx",
          status: "Ready",
          lastUpdate: "2006-01-02T15:04:05-0700",
        },
      ];

      const fields = [
        {
          label: "Name",
          value: ({ name }) => <a href="/some_url">{name}</a>,
        },
        {
          label: "Status",
          value: (v) => v.status,
        },
        {
          label: "Last Updated",
          value: "lastUpdate",
        },
      ];

      const tree = renderer
        .create(
          withTheme(
            <DataTable sortFields={["name"]} fields={fields} rows={rows} />
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
