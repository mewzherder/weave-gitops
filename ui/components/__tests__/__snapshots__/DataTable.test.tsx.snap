// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable snapshots renders 1`] = `
.c1 {
  font-family: 'proxima-nova',Helvetica,Arial,sans-serif;
  font-size: 16px;
  font-weight: 400;
  text-transform: none;
  font-style: normal;
}

.c0 th {
  font-weight: bold;
}

<div
  className="c0"
>
  <div
    className="MuiTableContainer-root"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Status
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Last Updated
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <span
              className="c1"
              size="normal"
            >
              <a
                href="/some_url"
              >
                my-cool-app
              </a>
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <span
              className="c1"
              size="normal"
            >
              Ready
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <span
              className="c1"
              size="normal"
            >
              2006-01-02T15:04:05-0700
            </span>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <span
              className="c1"
              size="normal"
            >
              <a
                href="/some_url"
              >
                nginx
              </a>
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <span
              className="c1"
              size="normal"
            >
              Ready
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <span
              className="c1"
              size="normal"
            >
              2006-01-02T15:04:05-0700
            </span>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <span
              className="c1"
              size="normal"
            >
              <a
                href="/some_url"
              >
                podinfo
              </a>
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <span
              className="c1"
              size="normal"
            >
              Ready
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <span
              className="c1"
              size="normal"
            >
              2006-01-02T15:04:05-0700
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
